### ISSUE TITLE
Insight into why backport failed

### ISSUE THREAD

**Issue by jackkoenig** [Mergifyio/mergify-engine#1654]
Sometimes backports fail. For example https://github.com/chipsalliance/chisel3/pull/1706/checks?check_run_id=1573373921 and then trying to do it manually: https://github.com/chipsalliance/chisel3/pull/1706#issuecomment-747740699.

The error report is just `Backport to branch <branch> failed`. Is there any way to get more insight/information about why it failed?

**Comment by jd** [comment]
Yes, this is due to conflict (or maybe empty patch). We don't post the full `git cherry-pick` log though we could!

**Comment by kymikoloco** [comment]
Just had a failure when a coworker commented `@mergifyio backport branch` before it was merged with this rule:
```yaml
    actions:
      merge:
        strict: true
        strict_method: rebase
        method: merge
```

> Command backport branch: success
> 
> > Backports have been created
> > 
> > * Backport to branch **branch** not needed, change already in branch **branch**

I commented `@mergifyio backport branch` after it was merged, which then succeeded in creating the backport PR. The current error for `DuplicateNotNeeded`  may not describe exactly what was happening or why it failed. Maybe adding the reason on to this error might help a bit more?

https://github.com/Mergifyio/mergify-engine/blob/8d8f9c09504e6667ef5dd904fff7c1e19a7b404f/mergify_engine/actions/copy.py#L163-L167
